sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -0,0 +1,449 @@
+
+  def load_model(self):
+    """Loads and initializes the model for processing."""
+    model = self._model_class.from_pretrained(
+        self._model_uri, **self._model_config_args)
+    if is_gpu_available_torch(self._device):
+      model.to(torch.device('cuda'))
+    return model
+
+  def update_model_path(self, model_path: Optional[str] = None):
+    self._model_uri = model_path if model_path else self._model_uri
+
+  def get_num_bytes(
+      self, batch: Sequence[Union[tf.Tensor, torch.Tensor]]) -> int:
+    """
+    Returns:
+      The number of bytes of data for the Tensors batch.
+    """
+    if self._framework == "tf":
+      return sum(sys.getsizeof(element) for element in batch)
+    else:
+      return sum(
+          (el.element_size() for tensor in batch for el in tensor.values()))
+
+  def batch_elements_kwargs(self):
+    return self._batching_kwargs
+
+  def share_model_across_processes(self) -> bool:
+    return self._large_model
+
+
+class HuggingFaceModelHandlerKeyedTensor(
+    HuggingFaceModelHandler[Dict[str, Union[tf.Tensor, torch.Tensor]],
+                            PredictionResult,
+                            Union[AutoModel, TFAutoModel]]):
+  """Implementation of the ModelHandler interface for HuggingFace with
+    Keyed Tensors for PyTorch/Tensorflow backend.
+
+    Depending on the type of tensors,
+    the model framework is determined automatically.
+
+    Example Usage model::
+    pcoll | RunInference(HuggingFaceModelHandlerKeyedTensor(
+      model_uri="bert-base-uncased", model_class=AutoModelForMaskedLM))
+
+  Args:
+    model_uri (str): path to the pretrained model on the hugging face
+      models hub.
+    model_class: model class to load the repository from model_uri.
+    device: For torch tensors, specify device on which you wish to
+      run the model. Defaults to CPU.
+    inference_fn: the inference function to use during RunInference.
+      Default is _run_inference_torch_keyed_tensor or
+      _run_inference_tensorflow_keyed_tensor depending on the input type.
+    load_model_args (Dict[str, Any]): keyword arguments to provide load
+      options while loading models from Hugging Face Hub. Defaults to None.
+    inference_args ([Dict[str, Any]]): Non-batchable arguments
+      required as inputs to the model's inference function. Unlike Tensors in
+      `batch`, these parameters will not be dynamically batched.
+      Defaults to None.
+    min_batch_size: the minimum batch size to use when batching inputs.
+    max_batch_size: the maximum batch size to use when batching inputs.
+    large_model: set to true if your model is large enough to run into
+      memory pressure if you load multiple copies. Given a model that
+      consumes N memory and a machine with W cores and M memory, you should
+      set this to True if N*W > M.
+    kwargs: 'env_vars' can be used to set environment variables
+      before loading the model.
+
```

Comment:
Do we need to mention the documentation again here? Instead can we address the difference between base class and the current one?

Reply:
Just an idea. I am fine with having the doc string here as well.